[inductor] Add more cudagraph tree logging

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #170218
* #170217

Summary: Add logging to report the graph inputs info at the graph recording time. Here is an example of the new log,

```
torch/_inductor/cudagraph_trees.py:409] [__cudagraphs] Recording cudagraph tree for symint key 3
torch/_inductor/cudagraph_trees.py:2587] [__cudagraphs] Checkpointing cuda caching allocator state. Number of checkpoints 1
torch/_inductor/cudagraph_trees.py:2310] [__cudagraphs] Running warmup of function 1 (f)
torch/_inductor/cudagraph_trees.py:2264] [__cudagraphs] Recording function 1 (f) of graph recording id 1, inputs: [0]: int(3), [1]: Tensor(size=[3], stride=(1,), dtype=torch.float32, data_ptr=0x7F4763000200)
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo @mlazos

Differential Revision: [D88980120](https://our.internmc.facebook.com/intern/diff/D88980120)

desertfire committed:
[inductor] Add more cudagraph tree logging

Summary: Add logging to report the graph inputs info at the graph recording time. Here is an example of the new log,

```
torch/_inductor/cudagraph_trees.py:409] [__cudagraphs] Recording cudagraph tree for symint key 3
torch/_inductor/cudagraph_trees.py:2587] [__cudagraphs] Checkpointing cuda caching allocator state. Number of checkpoints 1
torch/_inductor/cudagraph_trees.py:2310] [__cudagraphs] Running warmup of function 1 (f)
torch/_inductor/cudagraph_trees.py:2264] [__cudagraphs] Recording function 1 (f) of graph recording id 1, inputs: [0]: int(3), [1]: Tensor(size=[3], stride=(1,), dtype=torch.float32, data_ptr=0x7F4763000200)
```

[ghstack-poisoned]

mlazos commented:
If you want, you can add more logs testing for this. In general though the tests just make sure the logs are outputting something and aren't fine-grained enough to detect if you add *more* logging. If you see failures you might need to increase the expected number of records in the test.

desertfire committed:
Update on "[inductor] Add more cudagraph tree logging"


Summary: Add logging to report the graph inputs info at the graph recording time. Here is an example of the new log,

```
torch/_inductor/cudagraph_trees.py:409] [__cudagraphs] Recording cudagraph tree for symint key 3
torch/_inductor/cudagraph_trees.py:2587] [__cudagraphs] Checkpointing cuda caching allocator state. Number of checkpoints 1
torch/_inductor/cudagraph_trees.py:2310] [__cudagraphs] Running warmup of function 1 (f)
torch/_inductor/cudagraph_trees.py:2264] [__cudagraphs] Recording function 1 (f) of graph recording id 1, inputs: [0]: int(3), [1]: Tensor(size=[3], stride=(1,), dtype=torch.float32, data_ptr=0x7F4763000200)
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo mlazos

Differential Revision: [D88980120](https://our.internmc.facebook.com/intern/diff/D88980120)

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Add more cudagraph tree logging"


Summary: Add logging to report the graph inputs info at the graph recording time. Here is an example of the new log,

```
torch/_inductor/cudagraph_trees.py:409] [__cudagraphs] Recording cudagraph tree for symint key 3
torch/_inductor/cudagraph_trees.py:2587] [__cudagraphs] Checkpointing cuda caching allocator state. Number of checkpoints 1
torch/_inductor/cudagraph_trees.py:2310] [__cudagraphs] Running warmup of function 1 (f)
torch/_inductor/cudagraph_trees.py:2264] [__cudagraphs] Recording function 1 (f) of graph recording id 1, inputs: [0]: int(3), [1]: Tensor(size=[3], stride=(1,), dtype=torch.float32, data_ptr=0x7F4763000200)
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo mlazos

Differential Revision: [D88980120](https://our.internmc.facebook.com/intern/diff/D88980120)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D88980120).

desertfire committed:
Update on "[inductor] Add more cudagraph tree logging"


Summary: Add logging to report the graph inputs info at the graph recording time. Here is an example of the new log,

```
torch/_inductor/cudagraph_trees.py:409] [__cudagraphs] Recording cudagraph tree for symint key 3
torch/_inductor/cudagraph_trees.py:2587] [__cudagraphs] Checkpointing cuda caching allocator state. Number of checkpoints 1
torch/_inductor/cudagraph_trees.py:2310] [__cudagraphs] Running warmup of function 1 (f)
torch/_inductor/cudagraph_trees.py:2264] [__cudagraphs] Recording function 1 (f) of graph recording id 1, inputs: [0]: int(3), [1]: Tensor(size=[3], stride=(1,), dtype=torch.float32, data_ptr=0x7F4763000200)
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo mlazos

Differential Revision: [D88980120](https://our.internmc.facebook.com/intern/diff/D88980120)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D88980120).

desertfire committed:
Update on "[inductor] Add more cudagraph tree logging"


Summary: Add logging to report the graph inputs info at the graph recording time. Here is an example of the new log,

```
torch/_inductor/cudagraph_trees.py:409] [__cudagraphs] Recording cudagraph tree for symint key 3
torch/_inductor/cudagraph_trees.py:2587] [__cudagraphs] Checkpointing cuda caching allocator state. Number of checkpoints 1
torch/_inductor/cudagraph_trees.py:2310] [__cudagraphs] Running warmup of function 1 (f)
torch/_inductor/cudagraph_trees.py:2264] [__cudagraphs] Recording function 1 (f) of graph recording id 1, inputs: [0]: int(3), [1]: Tensor(size=[3], stride=(1,), dtype=torch.float32, data_ptr=0x7F4763000200)
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo mlazos

Differential Revision: [D88980120](https://our.internmc.facebook.com/intern/diff/D88980120)

[ghstack-poisoned]

desertfire committed:
Update on "[inductor] Add more cudagraph tree logging"


Summary: Add logging to report the graph inputs info at the graph recording time. Here is an example of the new log,

```
torch/_inductor/cudagraph_trees.py:409] [__cudagraphs] Recording cudagraph tree for symint key 3
torch/_inductor/cudagraph_trees.py:2587] [__cudagraphs] Checkpointing cuda caching allocator state. Number of checkpoints 1
torch/_inductor/cudagraph_trees.py:2310] [__cudagraphs] Running warmup of function 1 (f)
torch/_inductor/cudagraph_trees.py:2264] [__cudagraphs] Recording function 1 (f) of graph recording id 1, inputs: [0]: int(3), [1]: Tensor(size=[3], stride=(1,), dtype=torch.float32, data_ptr=0x7F4763000200)
```

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy kadeng muchulee8 amjames chauhang aakhundov coconutruben jataylo mlazos

Differential Revision: [D88980120](https://our.internmc.facebook.com/intern/diff/D88980120)

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge
